core/trino-main/src/main/java/io/trino/sql/planner/plan/TableWriterNode.java

```diff
@@ -208,6 +211,11 @@ public PlanNode replaceChildren(List<PlanNode> newChildren)
     {
         @Override
         public abstract String toString();
+
+        public boolean supportsReportingWrittenBytes(Metadata metadata, Session session)
+        {
+            return false;
```

Comment:
I would refrain from providing a default implementation. It could be error prone. Please return `false` where it is necessary or throw an `UnsupportedOperationException` when it is not expected to be called.